pytest_ethereum/deployer.py

```diff
@@ -0,0 +1,35 @@
+from eth_utils import to_canonical_address
+
+from ethpm import Package
+
+
+class Deployer:
+    def __init__(self, package):
+        if not isinstance(package, Package):
+            raise TypeError(
+                "Expected a Package object, instead received {0}.".format(type(package))
+            )
+        self.package = package
+
+    def deploy(self, contract_type, *args):
+        """
+        todo: chain-matching and use existing deployment as link_ref
+        """
+        factory = self.package.get_contract_factory(contract_type)
+        if factory.has_linkable_bytecode():
+            return self.deploy_linked_factory(factory, contract_type, args)
```

Comment:
I think automatic linking is a bad idea.  It's very difficult to get right in my experience.